hedera-mirror-importer/src/main/java/com/hedera/mirror/importer/parser/record/transactionhandler/NodeStakeUpdateTransactionHandler.java

```diff
@@ -0,0 +1,115 @@
+package com.hedera.mirror.importer.parser.record.transactionhandler;
+
+/*-
+ * ‌
+ * Hedera Mirror Node
+ * ​
+ * Copyright (C) 2019 - 2022 Hedera Hashgraph, LLC
+ * ​
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ * ‍
+ */
+
+import static com.hedera.mirror.common.util.DomainUtils.TINYBARS_IN_HBARS;
+
+import com.hederahashgraph.api.proto.java.NodeStakeUpdateTransactionBody;
+import java.time.LocalDate;
+import java.time.ZoneId;
+import java.util.Map;
+import java.util.concurrent.atomic.AtomicLong;
+import java.util.function.Function;
+import java.util.stream.Collectors;
+import javax.inject.Named;
+import lombok.RequiredArgsConstructor;
+
+import com.hedera.mirror.common.domain.addressbook.NodeStake;
+import com.hedera.mirror.common.domain.entity.EntityId;
+import com.hedera.mirror.common.domain.transaction.RecordItem;
+import com.hedera.mirror.common.domain.transaction.Transaction;
+import com.hedera.mirror.common.domain.transaction.TransactionType;
+import com.hedera.mirror.common.util.DomainUtils;
+import com.hedera.mirror.importer.parser.record.entity.EntityListener;
+import com.hedera.mirror.importer.repository.NodeStakeRepository;
+import com.hedera.mirror.importer.util.Utility;
+
+@Named
+@RequiredArgsConstructor
+class NodeStakeUpdateTransactionHandler implements TransactionHandler {
+
+    static final ZoneId ZONE_UTC = ZoneId.of("UTC");
+    private static final NodeStake EMPTY_NODE_STAKE = new NodeStake();
+
+    private final EntityListener entityListener;
+    private final NodeStakeRepository nodeStakeRepository;
+
+    @Override
+    public EntityId getEntity(RecordItem recordItem) {
+        return null;
+    }
+
+    @Override
+    public TransactionType getType() {
+        return TransactionType.NODESTAKEUPDATE;
+    }
+
+    @Override
+    public void updateTransaction(Transaction transaction, RecordItem recordItem) {
+        var transactionBody = recordItem.getTransactionBody().getNodeStakeUpdate();
+        long epochDay = toEpochDay(recordItem);
+        var previousStake = getPreviousStake(epochDay);
+        long rewardRate = transactionBody.getRewardRate();
+        long stakeTotal = getStakeTotal(transactionBody);
```

Comment:
Q: Why calculate here and iterate over `transactionBody.getNodeStakeList()` here only to do it again a few lines down?
Then in the for look we can sum the stake

```suggestion
        var stakeTotal = new AtomicLong(0L);
```

Reply:
Because you have to have the total stake to calculate the reward sum inside the loop. It requires you to loop twice

Reply:
Understood